tchannel/tornado/peer.py

```diff
@@ -618,7 +617,8 @@ def send_with_retry(self, request, peer, retry_limit, connection):
             try:
                 response = yield self._send(connection, request)
                 raise gen.Return(response)
-            except (ProtocolError, TimeoutError) as error:
+            # Why are we retying on all errors????
+            except TChannelError as error:
```

Comment:
The exception classes still have a `code` attribute on them if you need it.

The question here was because we're catching `ProtocolError` which all of our exceptions were inheriting from... it's still preserving that behavior by catching `TChannelError` (which is what everything now inherits from).